src/graphql/user.ts

```diff
@@ -1,18 +1,27 @@
-import { extendType, inputObjectType, objectType } from 'nexus';
+import { extendType, inputObjectType, nonNull, objectType } from 'nexus';
+import { User } from '../entity/User';
 
 export const CreateUser = extendType({
   type: 'Mutation',
   definition(t) {
     t.nonNull.field('createUser', {
       type: CreateUserResponse,
       args: {
-        data: UserInput,
+        data: nonNull(UserInput),
       },
 
-      resolve(parent, args, context) {
-        const { name, email, birthDate } = args.data;
+      resolve: async (_parent, args, context) => {
+        const { name, email, birthDate, password } = args.data;
 
-        return { id: 0, name, email, birthDate };
+        const user = new User();
+        user.name = name;
+        user.email = email;
+        user.birthDate = birthDate;
+        user.password = password;
+
+        const { id } = await context.typeorm.manager.save(user);
+
+        return { id, name, email, birthDate };
```

Comment:
O GraphQL já ajuda e coloca no retorno apenas os campos declarados no schema. Então vc pode retornar direto o resultado do save:


```suggestion
        return context.typeorm.manager.save(user);
```